storage/innobase/mtr/mtr0log.cc

```diff
@@ -799,9 +811,37 @@ bool mlog_open_and_write_index(mtr_t *mtr, const byte *rec,
     n = DICT_INDEX_SPATIAL_NODEPTR_SIZE;
   }
 
+  /* Ordinal position of an existing field can't be changed with INSTANT
+  algorithm. But when it is combined with ADD/DROP COLUMN, ordinal position
+  of a filed can be changed. This bool array of size #fields in index,
+  represents if ordinal position of an existing filed is changed. */
+
+  // Move the logic to find columns that changed order before calculating
+  // the buffer size
+  std::unique_ptr<bool[]> fields_with_changed_order = nullptr;
+  if (is_versioned) {
+    fields_with_changed_order = std::make_unique<bool[]>(n);
+    memset(fields_with_changed_order.get(), 0, (sizeof(bool) * n));
```

Comment:
not needed, make_unique default-initializes the array.

Reply:
Done.